### PR TITLE
Update README 

### PR DESCRIPTION
**Pull request by Jaclynpqc**
The purpose of this pull request is to enhance the README file by organizing sections, providing a detailed comparison between Chasten and Symbex, suggesting similar tools, and offering a quick reference for Chasten command options. These changes aim to improve user understanding, facilitate tool selection, and enhance overall documentation.

**Changes Made:**

- Section Organization: Organized sections in the README file for better clarity and readability.

- Chasten vs. Symbex Comparison Section: Added a new section titled `🤓 Chasten vs. Symbex` to provide a detailed comparison between Chasten and Symbex. This section covers their command-line interfaces (CLI) and functionality, helping users make informed decisions based on their preferences and requirements.

- Similar Tools Section: Introduced a section titled `📦 Similar Tools` that highlights alternative tools for code analysis. Included brief descriptions of two tools: pyastgrep by Luke Plant and treesitter.

- chasten--help Section: Added a new section titled `✨chasten--help` to showcase the available options when using the chasten command. This provides users with quick reference information on how to interact with Chasten through the command line.

### CODE REVIEW

**Comment by AlishChhetri** [comment]
This looks great, but I also have a few additions/ changes to the readme on a file I have not PRed yet. Would I be able to wrap that up and jump onto this PR, or should I just make another? I will have my edits in for review by the lab.

**Comment by AlishChhetri** [comment]
@Jaclynpqc Its about adding a section dedicated to testing and additional details about setting up the developer environment. Most of the content will have a focus on providing more details for future developers of the tool.

**Review by boulais01 — APPROVED** [review]

**Comment by laurennevill** [comment]
If its okay with @Jaclynpqc, lets have @AlishChhetri jump on and add his README changes so that we can get all changes pushed at once!

File: README.md [hunk 1]
```diff
@@ -345,8 +327,67 @@ class DebugLevel(str, Enum):
...
+
+```
+
+## 🧑‍💻 Development Enviroment
 
+### 🏠 Local
 
+Follow these steps to install the `chasten` tool for future development:
+
+- The developement and use of Chasten requires [Python 3.11](https://www.python.org/downloads/release/python-3115/), must be greater or equal to version 3.11.5.
```
Comment (laurennevill) [inline README.md:359]: minor typo: development*

**Review by laurennevill — CHANGES_REQUESTED** [review]

**Review by laurennevill — APPROVED** [review]

**Review by AlishChhetri — DISMISSED** [review]
Looks great!

**Pull request merged**